mlrun/datastore/sources.py

```diff
@@ -428,6 +423,7 @@ def to_dataframe(
         additional_filters=None,
     ):
         reader_args = self.attributes.get("reader_args", {})
+        additional_filters = transform_list_filters_to_tuple(additional_filters)
```

Comment:
8 lines down, we have `additional_filters=additional_filters or self.additional_filters`. What if we end up getting the additional filters from self and not from the parameter?

Also, this is preexisting, but I wonder why here it's `additional_filters or self.additional_filters` and in `to_step()` it's `self.additional_filters or additional_filters`.

Reply:
about 1, same explanation as my comment before.
2 - you are absolutely right, but this are the convention of other parameters in these function, If I will change them I can break the behavior.